From bfc9fbaed6f1d43601d10d1899a1e26f42f1afab Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Tue, 26 Feb 2013 18:47:47 +0200
Subject: ARM: i.MX6: cm-fx6: fix the SD3 CD and WP

sb-fx6 has its card detect pin on GPIO7_1 and write protect on GPIO7_0.
Adjust the pin mux and platform data to fix the SD3 CD and WP
functionality.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |    7 ++++---
 arch/arm/mach-mx6/board-cm-fx6-q.h  |    7 ++++---
 arch/arm/mach-mx6/board-cm-fx6.c    |   13 +++++++------
 3 files changed, 15 insertions(+), 12 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index a6f7d3c..e2ab0148 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -69,9 +69,10 @@ static iomux_v3_cfg_t cm_fx6_dl_pads[] = {
 	MX6DL_PAD_SD3_DAT1__USDHC3_DAT1_50MHZ,
 	MX6DL_PAD_SD3_DAT2__USDHC3_DAT2_50MHZ,
 	MX6DL_PAD_SD3_DAT3__USDHC3_DAT3_50MHZ,
-	/* SD3_CD and SD3_WP */
-	MX6DL_PAD_NANDF_CS0__GPIO_6_11,
-	MX6DL_PAD_NANDF_CS1__GPIO_6_14,
+	/* SB-FX6 SD3 WP and CD */
+	MX6DL_PAD_SD3_DAT5__GPIO_7_0,
+	MX6DL_PAD_SD3_DAT4__GPIO_7_1,
+
 	/* eCSPI1 */
 	MX6DL_PAD_EIM_EB2__ECSPI1_SS0,
 	MX6DL_PAD_EIM_D16__ECSPI1_SCLK,
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index eb2fb54..cd428e6 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -69,9 +69,10 @@ static iomux_v3_cfg_t cm_fx6_q_common_pads[] = {
 	MX6Q_PAD_SD3_DAT1__USDHC3_DAT1_50MHZ,
 	MX6Q_PAD_SD3_DAT2__USDHC3_DAT2_50MHZ,
 	MX6Q_PAD_SD3_DAT3__USDHC3_DAT3_50MHZ,
-	/* SD3_CD and SD3_WP */
-	MX6Q_PAD_NANDF_CS0__GPIO_6_11,
-	MX6Q_PAD_NANDF_CS1__GPIO_6_14,
+	/* SB-FX6 SD3 WP and CD */
+	MX6Q_PAD_SD3_DAT5__GPIO_7_0,
+	MX6Q_PAD_SD3_DAT4__GPIO_7_1,
+
 	/* eCSPI1 */
 	MX6Q_PAD_EIM_EB2__ECSPI1_SS0,
 	MX6Q_PAD_EIM_D16__ECSPI1_SCLK,
diff --git a/arch/arm/mach-mx6/board-cm-fx6.c b/arch/arm/mach-mx6/board-cm-fx6.c
index d53dd3c..58badd1 100644
--- a/arch/arm/mach-mx6/board-cm-fx6.c
+++ b/arch/arm/mach-mx6/board-cm-fx6.c
@@ -64,6 +64,7 @@
 #include <mach/ipu-v3.h>
 #include <mach/mxc_hdmi.h>
 #include <mach/mxc_asrc.h>
+#include <mach/esdhc.h>
 
 #include <asm/irq.h>
 #include <asm/setup.h>
@@ -85,11 +86,12 @@
 #define CM_FX6_ECSPI1_CS1		IMX_GPIO_NR(3, 19)
 #define CM_FX6_USB_HUB_RST		IMX_GPIO_NR(7, 8)
 
+#define SB_FX6_SD3_WP			IMX_GPIO_NR(7, 0)
+#define SB_FX6_SD3_CD			IMX_GPIO_NR(7, 1)
+
 #define MX6_ARM2_LDB_BACKLIGHT		IMX_GPIO_NR(1, 9)
 #define MX6_ARM2_USB_OTG_PWR		IMX_GPIO_NR(3, 22)
 #define MX6_ARM2_CAN2_EN		IMX_GPIO_NR(5, 24)
-#define MX6_ARM2_SD3_CD			IMX_GPIO_NR(6, 11)
-#define MX6_ARM2_SD3_WP			IMX_GPIO_NR(6, 14)
 #define MX6_ARM2_CAN1_STBY		IMX_GPIO_NR(7, 12)
 #define MX6_ARM2_CAN1_EN		IMX_GPIO_NR(7, 13)
 
@@ -169,10 +171,9 @@ static int plt_sd_pad_change(unsigned int index, int clock)
 }
 
 static const struct esdhc_platform_data mx6_arm2_sd3_data __initconst = {
-	.cd_gpio		= MX6_ARM2_SD3_CD,
-	.wp_gpio		= MX6_ARM2_SD3_WP,
-	.support_18v		= 1,
-	.support_8bit		= 1,
+	.cd_type		= ESDHC_CD_GPIO,
+	.cd_gpio		= SB_FX6_SD3_CD,
+	.wp_gpio		= SB_FX6_SD3_WP,
 	.keep_power_at_suspend	= 1,
 	.delay_line		= 0,
 	.platform_pad_change	= plt_sd_pad_change,
-- 
1.7.9.5

